include/treelite/tree.h

```diff
@@ -424,18 +471,25 @@ struct Model {
   ModelParam param;
 
   /*! \brief disable copy; use default move */
-  Model() = default;
-  ~Model() = default;
-  Model(const Model&) = delete;
-  Model& operator=(const Model&) = delete;
-  Model(Model&&) = default;
-  Model& operator=(Model&&) = default;
-
-  void ReferenceSerialize(dmlc::Stream* fo) const;
-
-  inline std::vector<PyBufferFrame> GetPyBuffer();
-  inline void InitFromPyBuffer(std::vector<PyBufferFrame> frames);
-  inline Model Clone() const;
+  ModelImpl() = default;
+  ~ModelImpl() = default;
+  ModelImpl(const ModelImpl&) = delete;
+  ModelImpl& operator=(const ModelImpl&) = delete;
+  ModelImpl(ModelImpl&&) noexcept = default;
+  ModelImpl& operator=(ModelImpl&&) noexcept = default;
+
+  inline ModelParam GetParam() const override;
```

Comment:
`GetParam()`, `GetNumFeature()`, `GetNumOutputGroup()` and `GetRandomForestFlag()` could be defined in the base class if the corresponding fields are moved to the base class.

Reply:
Fixed in #201